js-intermediate.md

```diff
@@ -5,94 +5,92 @@ Answer the following questions. First, without external resources. Challenge you
 
 1. What are the HTTP verbs? Which CRUD action corresponds to each verb?
 
-  Your answer:
+  Your answer: post, patch, push, delete...
 
-  Researched answer:
+  Researched answer: get -> read, post -> create, push -> update, patch -> update, delete -> delete
 
 
-2. Why would you use object destructuring?
+2. Why would you use object destructuring? 
 
-  Your answer:
+  Your answer: To shorten code and make it easier to read.
 
-  Researched answer:
+  Researched answer: "Destructuring can make your code more readable and maintainable by reducing the line count and making the intent clear. Array destructuring can allow you to use arrays as though they were tuples."
 
 
 
 3. Why is testing important?
 
-  Your answer:
+  Your answer: So you can make sure the application is running properly.
 
-  Researched answer:
+  Researched answer: "A product undergoes multiple levels of a Testing method to ensure its reliable functioning. It is the most important step in the process of product development. ... It helps to identify any defects or errors that could possibly be made during the development phase."
 
 
 4. What is the difference between a class and an object?
 
-  Your answer:
+  Your answer: An object contains the class ?
 
-  Researched answer:
+  Researched answer: What I gathered from the syllabus: classes define attributes and behaviors, while objects interact with each other and make up an application.
 
```

Comment:
An object is an instance of a class. Objects and classes have data and behavior.